util/chplenv/chpl_platform.py

```diff
@@ -40,11 +38,11 @@ def get(flag='host'):
                     platform_val = 'cray-xe'
                 elif net.lower() == 'ari':
                     platform_val = 'cray-xc'
-            if not platform_val:
-                product_pattern = re.compile(r'^PRODUCT=.*\b[Ss]hasta\b', re.MULTILINE)
-                product_match = product_pattern.search(cle_info)
-                if product_match is not None:
-                    platform_val = 'cray-shasta'
+
+    if not platform_val:
+        network = os.environ.get('CRAYPE_NETWORK_TARGET')
```

Comment:
This should be `os.environ.get('CRAYPE_NETWORK_TARGET', 'none')` or `os.environ.get('CRAYPE_NETWORK_TARGET', '')` (default if the env var is unset is `None`, not a string

Reply:
perils of blindly following the `CHPL_BUILD_X86_64_AS_32` later in the file, but not realizing the effect on the `startswith()` method ... will fix